### PR TITLE
Add New Formatted heatmap

### LINKED ISSUE #24: Add New Formatted heatmap

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
Fixes #24

Here is the video to show the time taken to produce the results:

https://github.com/PolicyEngine/us-marriage-incentive/assets/73863365/ebeaf483-cc54-45a0-99cb-4ec8e67873db

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
Can we invert the y-axis so both x and y start at 0 in the bottom left. 

Could we also make the values uniform 10k to 100k in 10k steps for both x and y 

Additionally could you add "Penalty" to the legend on the right

File: app.py [hunk 1]
```diff
@@ -94,6 +95,13 @@ def get_net_income(state_code, head_employment_income, spouse_employment_income=
     situation["households"] = {
         "your household": {"members": members, "state_name": {"2023": state_code}}
     }
+    # situation["axes"] = [[ {
+    #     "name": "employment_income",
+    #     "count": 200,
+    #     "min": 0,
+    #     "max": 200000
+    #   }]]
```
Comment (PavelMakarchuk) [inline app.py:103]: delete commented code

**Comment by MaxGhenis** [comment]
Could we add a second button to generate the heatmap? (With a note saying it takes 2 minutes.) Currently the user won't know to wait for it.

**Comment by ThapeloMasasa** [comment]
> Could we add a second button to generate the heatmap? (With a note saying it takes 2 minutes.) Currently the user won't know to wait for it.

Here is a spinner with a message
https://github.com/PolicyEngine/us-marriage-incentive/assets/73863365/157e9c47-3232-49d4-a113-ef274873f61b

**Comment by MaxGhenis** [comment]
Nice thanks. Please make the heatmap have color gradations so that it aligns with the legend. And let's use the PolicyEngine color palette from [here](https://github.com/PolicyEngine/policyengine-app/blob/6bd2ce249ad5752b50ba6c37982b10d379813bc4/src/style/colors.jsx#L2): blue for bonuses, gray for penalties.

**Comment by ThapeloMasasa** [comment]
Here is the changed heatmap:
Screenshot 2024-03-11 at 8 19 25 AM

**Comment by PavelMakarchuk** [comment]
Looks great, is it possible for the scale to go from -1 to 1 with 0 being no penalty or bonus (seems more intutive)?

**Comment by PavelMakarchuk** [comment]
need to resolve the conflict

**Comment by ThapeloMasasa** [comment]
Updated heat map

Screenshot 2024-03-13 at 11 51 17 AM

**Comment by PavelMakarchuk** [comment]
Could make the scale from -1 over 0 to 1 (1 Bonus, 0 being neutral and 1 max penalty) - dont need the labels for now

**Comment by ThapeloMasasa** [comment]
> Could make the scale from -1 over 0 to 1 (1 Bonus, 0 being neutral and 1 max penalty) - dont need the labels for now

the colorscale property takes in normalized values so they have to be within the range of 0-1.. negative values cause  an error

**Pull request merged**